sections/nix_shell.zsh

```diff
@@ -0,0 +1,40 @@
+#
+# Nix Shell
+#
+# Nix can be used to provide some kind of virtual environment through the nix-shell command.
+# Link: https://nixos.org/manual/nix/stable/command-ref/nix-shell.html
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+SPACESHIP_NIXSHELL_SHOW="${SPACESHIP_NIXSHELL_SHOW=true}"
+SPACESHIP_NIXSHELL_PREFIX="${SPACESHIP_NIXSHELL_PREFIX="$SPACESHIP_PROMPT_DEFAULT_PREFIX"}"
+SPACESHIP_NIXSHELL_SUFFIX="${SPACESHIP_NIXSHELL_SUFFIX="$SPACESHIP_PROMPT_DEFAULT_SUFFIX"}"
+SPACESHIP_NIXSHELL_SYMBOL="${SPACESHIP_NIXSHELL_SYMBOL="❄ "}"
+SPACESHIP_NIXSHELL_COLOR="${SPACESHIP_NIXSHELL_COLOR="yellow"}"
+
+# ------------------------------------------------------------------------------
+# Section
+# ------------------------------------------------------------------------------
+
+# Shows whether nix-shell environment is active
+spaceship_nixshell() {
+  [[ $SPACESHIP_NIXSHELL_SHOW == false ]] && return
+
+  [[ -z $IN_NIX_SHELL ]] && return
+
+  if [[ -z $name || $name = "" ]] then
```

Comment:
Seems, like the right side of the `||` expression would not work as expected.

```suggestion
  if [[ -z $name || $name == "" ]] then
```

Reply:
Hm, are you sure? It worked fine for me.

The zsh docs say both are valid:
```
string = pattern
string == pattern
true if string matches pattern. The two forms are exactly equivalent. The ‘=’ form is the traditional shell syntax (and hence the only one generally used with the test and [ builtins); the ‘==’ form provides compatibility with other sorts of computer language.
```

Source: https://zsh.sourceforge.io/Doc/Release/Conditional-Expressions.html

I'm happy to change it to `==` if there is some benefit that I'm not aware of

Reply:
Please, replace to `==` as it would be more familiar for the rest of the contributors (including me).

Reply:
Done 👍